### PR TITLE
Shift responsibility for keeping Date header to ResponseHeaderBag

### PR DESCRIPTION
**Pull request by mpdude**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

This is an improvement over #22036. It shifts responsibility for preserving a `Date` header to the `ResponseHeaderBag`.

We already have similar logic there for the `Cache-Control` header.

### CODE REVIEW

**Comment by fabpot** [comment]
I would do that in master as there is no bug fix here.

**Comment by mpdude** [comment]
@fabpot Fine for me. Can you do this when merging, or do you need a new PR?

**Comment by fabpot** [comment]
You can change the target branch on the PR directly.

**Comment by mpdude** [comment]
Must be a new GH feature or it has always been so well hidden I missed it.

Changed.

File: src/Symfony/Component/HttpFoundation/Tests/ResponseHeaderBagTest.php
Comment (ro0NL) [inline src/Symfony/Component/HttpFoundation/Tests/ResponseHeaderBagTest.php]: Im not sure you can simply remove the cache-control expectations, perhaps re-add it in a separate test. Which is probably best in the long run.

Comment (mpdude) [inline src/Symfony/Component/HttpFoundation/Tests/ResponseHeaderBagTest.php]: If I am not mistaken, there are other tests for them. In this test method that was only noise AFAICT, but please correct me if I'm wrong

Comment (ro0NL) [inline src/Symfony/Component/HttpFoundation/Tests/ResponseHeaderBagTest.php]: You're right :+1: same situation is covered in `testCacheControlHeader`

Comment (mpdude) [inline src/Symfony/Component/HttpFoundation/Tests/ResponseHeaderBagTest.php]: Thanks!

**Comment by mpdude** [comment]
@fabpot Does that mean you've postponed the merge from 3.3 to 3.4 :-(?

**Comment by mpdude** [comment]
@fabpot Anything left here I could do?

**Review by fabpot — APPROVED** [review]
Can you rebase before I merge to get rid of the merge commits? Thanks.

**Comment by fabpot** [comment]
If you could also change the target to 3.4, that would be wonderful.

**Comment by mpdude** [comment]
@fabpot changed base to 3.4 – did I get you right?

Also, at least in the web interface, GitHub shows no merge conflicts...?

**Comment by fabpot** [comment]
3.4 👍 
Can you still rebase please? We don't merge PRs with merge commits.

**Comment by mpdude** [comment]
Hm, rebasing and fixing up a branch that I had previously rebased in the GH web interface, including merge conflicts, really challenged my git skills... hope it's allright now.

**Comment by fabpot** [comment]
Apparently, tests are broken on deps low.

**Comment by mpdude** [comment]
Messed up the test during merge conflict resolution, should be fixed now.

**Comment by fabpot** [comment]
Thank you @mpdude.

**Pull request closed without merging**
